doc/howtos/rdtraining/setup.rst

```diff
@@ -0,0 +1,429 @@
+    $ ssh-add ~/.ssh/id_ed25519
+    $ sudo apt-get install xclip
+    $ xclip -sel clip < ~/.ssh/id_ed25519.pub
+
+
+On Github:
+
+
+- In the upper-right corner of any page, click your profile photo, then click Settings
+
+  .. image:: setup/media/userbar-account-settings.png
+
+- In the user settings sidebar, click SSH and GPG keys.
+
+  .. image:: setup/media/settings-sidebar-ssh-keys.png
+
+- Click New SSH key or Add SSH key.
+
+  .. image:: setup/media/ssh-add-ssh-key.png
+
+- In the "Title" field, add a descriptive label for the new key.
+- Paste your key into the "Key" field.
+
+  .. image:: setup/media/ssh-key-paste.png
+
+- Click Add SSH key.
+
+
+Fetch the sources
+-----------------
+
+All the Odoo sources will be located in `$HOME/src/`
+
+.. code-block:: console
+
+    $ mkdir -p $HOME/src
+    $ cd $HOME/src
+    $ git clone git@github.com:odoo/odoo.git
+    $ git clone git@github.com:odoo/enterprise.git
+
+.. tip:: Cloning the repositories will take a while, enjoy a cup of coffee while waiting.
+
+.. tip:: You may need to ask your manager read rights to fetch the enterprise repository.
+
+.. _howto/rdtraining/setup/development_repository:
+
+Configure development repository
+--------------------------------
+
+To contribute to Odoo's development, you will need to
+`fork the repository <https://guides.github.com/activities/forking/>`__, create a branch containing
+your code in the fork and submit a
+`Pull Request <https://docs.github.com/en/github/getting-started-with-github/github-glossary#pull-request>`__
+to the Odoo repository.
+
+In the case you are lucky enough to work at Odoo, the forks already exist. They are called
+`odoo-dev/odoo` and `odoo-dev/enterprise`.
+
+.. code-block:: console
+
+    $ cd  $HOME/src/odoo
+    $ git remote add odoo-dev git@github.com:odoo-dev/odoo.git #add odoo-dev as a new remote
+    $ git remote rename origin odoo #change the name of origin (the odoo repository) to odoo
+    $ git remote set-url --push odoo no_push #remove the possibility to push to odoo (you can only push to odoo-dev)
+
+    $ cd  $HOME/src/enterprise
+    $ git remote add enterprise-dev git@github.com:odoo-dev/enterprise.git
+    $ git remote rename origin enterprise
+    $ git remote set-url --push enterprise no_push
+
+
+Useful git commands
+-------------------
+
+These are useful git commands for your day-to-day work.
+
+* Change branch:
+  When you change branches, both repositories (odoo and enterprise) must be synchronized, i.e. both
+  need to be in the same branch.
+
+  .. code-block:: console
+
+    $ cd $HOME/src/odoo
+    $ git checkout 14.0
+
+    $ cd $HOME/src/enterprise
+    $ git checkout 14.0
+
+* Fetch and rebase:
+
+  .. code-block:: console
+
+    $ cd $HOME/src/odoo
+    $ git fetch --all --prune
+    $ git rebase --autostash odoo/14.0
+
+    $ cd $HOME/src/enterprise
+    $ git fetch --all --prune
+    $ git rebase --autostash enterprise/14.0
+
+
+Install the dependencies
```

Comment:
I see the added value of a single-page setup doc, as well as comments and tips included in that doc, but I believe that making a copy of the source install doc is a bad idea. A quick reference to the targeted page would do the trick too: "Refer to the Source Install documentation to prepare your environment."
As that page needs to be continuously updated along the year (not only when we release a major version), this training page will certainly be quickly outdated and confuse the readers more than it will help them.
Same goes for removing some useful bits as the command to install the C dependencies on Debian. This can be quite a challenge to figure it out for a new dev.

Reply:
> A quick reference to the targeted page would do the trick too

This page has been written for a reason: a large percentage of newcomers have a badly configured system after their training.

The point of this part of the tutorial is to install the git sources of Odoo 14.0 with wkhtml 0.12.5 on a clean Ubuntu 20.04.
The point of the "Source Install" on the documentation is to be generic for Linux installation. IMHO, the latter misses the point since:
- it assumes the system uses apt, so some kind of Debian derivative
- even by assuming a Debian derivative it doesn't target any version

Documenting how to install Odoo on a generic Linux distribution is simply impossible. Even by making the assumption the distribution is based on Debian you can never be sure the instructions will work for any version : some dependencies may or may not be already installed [1], some packages may not be available...

Moreover, you still have the possibility to not use git but download the sources in a zip file, install the multiple versions of wkhtml, whether or not use the Enterprise repo. When you start from scratch, the correct steps to take are not obvious.

In end, it's likely that the instructions in the documentation won't be enough and a few extra steps will be necessary to launch Odoo. We want to avoid this for newcomers: we want a step-by-step procedure which works every time.

> this training page will certainly be quickly outdated and confuse the readers more than it will help them.

The procedure is likely to be executed multiple times every 2 weeks, so hopefully we will be able to keep it up-to-date.

[1] A clean Debian 10 server has not the same set of pre-installed packages than a clean Ubuntu 20.04 or Ubuntu 18.04.